test/integration/agent-integration.test.ts

```diff
@@ -233,5 +233,29 @@ describe('Integration test', () => {
 
       })
     })
+
+    describe('canvas', () => {
+      it('captures canvas elements', async () => {
+        await page.goto(`http://localhost:${PORT}/serialize-canvas.html`)
+        await page.waitFor('#webgl canvas')
+        // I cannot think of a nicer way to let the canvas animations/drawing settle
+        // so sadly, use a timeout
+        await page.waitFor(1000)
+        const domSnapshot = await snapshot(page, 'Canvas elements')
+        const $ = cheerio.load(domSnapshot)
+
+        expect($('[data-percy-canvas-serialized]').length).to.equal(2)
+      })
+
+      it("doesn't serialize with JS enabled", async () => {
+        await page.goto(`http://localhost:${PORT}/serialize-canvas.html`)
+        await page.waitFor('#webgl canvas')
+        await page.waitFor(1000)
+        const domSnapshot = await snapshot(page, 'Canvas elements -- with JS', { enableJavaScript: true })
+        const $ = cheerio.load(domSnapshot)
+
+        expect($('[data-percy-canvas-serialized]').length).to.equal(0)
```

Comment:
Now I'm being nitpicky

```suggestion
        expect($('[data-percy-canvas-serialized]')).to.have.lengthOf(0)
```